src/transformers/pipelines.py

```diff
@@ -1180,19 +1206,69 @@ def __call__(
 
         """
         assert return_tensors or return_text, "You must specify return_tensors=True or return_text=True"
-        if self.framework == "tf":
-            raise NotImplementedError("Tensorflow not supported")
+        is_t5 = "T5ForConditionalGeneration" in self.model.__class__.__name__
+        is_bart = "BartForConditionalGeneration" in self.model.__class__.__name__
+
+        if self.framework == "tf" and is_bart:
+            raise NotImplementedError(
+                "Tensorflow is not yet supported for Bart. Please consider using T5, e.g. `t5-base`"
+            )
+
+        if is_t5:
+            assert len(documents) > 0, "Please provide a document to summarize"
+            if isinstance(documents[0], str):
+                documents = (T5_PREFIX_PATTERNS["summarization"] + documents[0],)
+            elif isinstance(documents[0], list):
+                documents = ([T5_PREFIX_PATTERNS["summarization"] + document for document in documents[0]],)
+
+        if isinstance(documents[0], list):
+            assert (
+                self.tokenizer.pad_token_id is not None
+            ), "Please make sure that the tokenizer has a pad_token_id when using a batch input"
+
         with self.device_placement():
-            inputs = self._parse_and_tokenize(*documents)
-            inputs = self.ensure_tensor_on_device(**inputs)
+            inputs = self._parse_and_tokenize(*documents, pad_to_max_length=True)
+
+            if self.framework == "pt":
+                inputs = self.ensure_tensor_on_device(**inputs)
+                input_length = inputs["input_ids"].shape[-1]
+            elif self.framework == "tf":
+                input_length = tf.shape(inputs["input_ids"])[-1]
```

Comment:
Ok for now but we'll probably refactor this in the future to have framework-agnostic `ensure_tensor_on_device` and `get_tensor_length` methods (maybe have a base class and framework-specific derived class for instance).